modules/jenkins.js

```diff
@@ -0,0 +1,90 @@
+// -----------------------------------------------------------------------------------------
+// #region Imports
+// -----------------------------------------------------------------------------------------
+
+const {
+    StringUtils,
+    CollectionUtils,
+} = require("andculturecode-javascript-core");
+const echo = require("./echo");
+const fs = require("fs");
+const os = require("os");
+const path = require("path");
+const upath = require("upath");
+
+// #endregion Imports
+
+// -----------------------------------------------------------------------------------------
+// #region Constants
+// -----------------------------------------------------------------------------------------
+const CONFIG_FILE = ".jenkinsconfig";
+const BASE_CONFIG = {
+    url: "",
+    username: "",
+    token: "",
+    profiles: {},
+};
+
+// #endregion Constants
+
+// -----------------------------------------------------------------------------------------
+// #region Functions
+// -----------------------------------------------------------------------------------------
+
+const jenkins = {
+    configureCredentials(url, username, token) {
+        const errors = [];
+        if (StringUtils.isEmpty(url)) {
+            errors.push("Url is required");
+        }
+        if (StringUtils.isEmpty(username)) {
+            errors.push("Username is required");
+        }
+        if (StringUtils.isEmpty(token)) {
+            errors.push("Api token is required");
+        }
+        if (CollectionUtils.hasValues(errors)) {
+            echo.errors(errors);
+            return false;
+        }
+        let config = BASE_CONFIG;
+        Object.assign(config, { url, username, token });
+        this.writeToConfig(config);
+        return true;
+    },
+    getConfig() {
+        const configPath = this.getConfigPath();
+        const configFile = fs.readFileSync(configPath);
+        if (configFile === null || configFile === undefined) {
+            return undefined;
+        }
+        if (StringUtils.isEmpty(configFile)) {
+            return BASE_CONFIG;
+        }
+        return JSON.parse(configFile);
+    },
+    getConfigPath() {
+        const homeDir = os.homedir();
+        const configPath = upath.toUnix(path.join(homeDir, CONFIG_FILE));
```

Comment:
I think this is another copy/paste thing that we need to go back & reconcile. After digging into `upath` further, it claims to be a full drop-in replacement for `path` - I think we can just leverage `upath.join` instead of calling `path.join` too.